pkg/logs/tailers/windowsevent/tailer_test.go

```diff
@@ -267,6 +267,177 @@ func (s *ReadEventsSuite) TestBookmarkNewTailer() {
 	// if tailer started from bookmark correctly, there should only be s.numEvents
 }
 
+// TestInitialBookmarkSeeding verifies the fix for the amnesia bug:
+// When a tailer starts with no bookmark, it creates an initial bookmark
+// from the most recent event and saves it immediately, even if no events
+// are processed before shutdown.
+func (s *ReadEventsSuite) TestInitialBookmarkSeeding() {
+	config := Config{
+		ChannelPath: s.channelPath,
+	}
+
+	// Step 1: Generate N=10 initial events to establish a baseline
+	// These events ensure there's a "most recent event" for initial seeding
+	initialEvents := uint(10)
+	err := s.ti.GenerateEvents(s.eventSource, initialEvents)
+	s.Require().NoError(err)
+
+	// Step 2: Start tailer with empty bookmark
+	// This triggers createInitialBookmark() which should:
+	// 1. Query for the most recent event
+	// 2. Create a bookmark from that event
+	// 3. Send a synthetic message to persist the bookmark
+	msgChan := make(chan *message.Message, 100) // Buffer to avoid blocking
+	tailer, err := newtailer(s.ti.API(), &config, "", msgChan)
+	s.Require().NoError(err)
+
+	// Step 3: Get the initial bookmark directly from the registry
+	// The implementation now uses direct SetOffset instead of synthetic messages
+	var initialBookmark string
+
+	// Wait a moment for the tailer to initialize and set the bookmark
+	time.Sleep(100 * time.Millisecond)
+
+	// Get the bookmark directly from the mock registry
+	mockRegistry := tailer.registry.(*auditormock.Auditor)
+	initialBookmark = mockRegistry.GetOffset(tailer.Identifier())
+	s.Require().NotEmpty(initialBookmark, "Initial bookmark must not be empty")
```

Comment:
avoid sleeps, different environments can have different delays and cause flaky failures. Instead, use a wait loop

```suggestion
        initialBookmark := ""
    	err := backoff.Retry(func() error {
            // Get the bookmark directly from the mock registry
            mockRegistry := tailer.registry.(*auditormock.Auditor)
            initialBookmark = mockRegistry.GetOffset(tailer.Identifier())
            if initialBookmark == "" {
                return fmt.Errorf("bookmark is empty")
            }
            return nil
	}, backoff.NewConstantBackOff(100*time.Millisecond))
	s.Require().NoError(err, "initial bookmark must not be empty")
```